portfolio/src/main/webapp/script.js

```diff
@@ -266,4 +266,37 @@ function buildInfoWindowInput(lat, lng) {
   containerDiv.appendChild(button);
 
   return containerDiv;
+}
+function authentication() {
```

Comment:
Non blocking comment: The bigger your file becomes the harder would it become to make sure everything is working as intended if you don't have tests.